code/modules/mob/living/carbon/human/species.dm

```diff
@@ -256,50 +257,61 @@ GLOBAL_LIST_EMPTY(features_by_species)
 	if(heart && (!should_have_heart || replace_current))
 		heart.Remove(C,1)
 		QDEL_NULL(heart)
+		required_organs -= heart
```

Comment:
This is creating hard delete potentials by using references and keeping them even after deletion.